ipywidgets/widgets/embed.py

```diff
@@ -0,0 +1,262 @@
+    return store
+
+
+def add_resolved_links(store, drop_defaults):
+    """Checks if any link models exists between models in store"""
+    for widget_id, widget in Widget.widgets.items(): # go over all widgets
+        if isinstance(widget, Link) and widget_id not in store:
+            if widget.source[0].model_id in store and widget.target[0].model_id in store:
+                store[widget.model_id] = widget._get_embed_state(drop_defaults=drop_defaults)
+
+
+def dependency_state(widgets, drop_defaults):
+    """Get the state of all widgets specified, and their dependencies.
+
+    In the below graph, D and E are depencies of C; C is a dependency of
+    both A and B; and E is a dependency of F. That means the state of C
+    will include (C, D, E).
+
+    A --           -- D
+        | -- C -- |
+    B --           --
+                     | -- E
+                 F --
+
+    ---- Dependecy ---->
+
+    Note: Any links between included widgets will also be added. Using the
+    example above, this means that any links between widgets (C, D, E) will
+    also be included.
+
+    """
+    # collect the state of all relevant widgets
+    if widgets is None:
+        # Get state of all widgets, no smart resolution needed.
+        widgets = Widget.widgets.values()
+        state = Widget.get_manager_state(drop_defaults=drop_defaults, widgets=widgets)['state']
+    else:
+        state = {}
+        for widget in widgets:
+            get_recursive_state(widget, state, drop_defaults)
+        # Add any links between included widgets:
+        add_resolved_links(state, drop_defaults)
+    return state
+
+
+def embed_data(views, include_all=True, drop_defaults=True):
+    """Gets data for embedding.
+
+    Use this to get the raw data for embedding if you have special
+    formatting needs.
+
+    Parameters
+    ----------
+    views: widget or collection of widgets or None
+        The widgets to include views for. If None, all DOMWidgets are
+        included (not just the displayed ones).
+    include_all: boolean
+        Which other widgets' state to include. When set to True, the state
+        of all widgets know to the widget manager is included. When False,
+        the dependencies of the given views are (attempted) resolved
+        automatically, and only their state is included.
+    drop_defaults: boolean
+        Whether to drop default values from the widget states.
+
+    Returns
+    -------
+    A dictionary with the following entries:
+        manager_state: dict of the widget manager state data
+        view_specs: a list of widget view specs
+    """
+    if views is not None:
+        try:
+            views[0]
+        except (IndexError, TypeError):
+            views = [views]
+    if include_all:
+        state = Widget.get_manager_state(drop_defaults=drop_defaults, widgets=None)['state']
+    else:
+        state = dependency_state(views, drop_defaults)
```

Comment:
While if only the target of the link is in store, the link should be added to the store, as well as the source of the link...

Then we must start over again with the rexursive wall from the source...